chaste_codegen/chaste_model.py

```diff
@@ -65,6 +71,16 @@ def __init__(self, model, file_name, **kwargs):
 
         self._model = model
 
+        # retrieve probabilities that don't stay in 0 ... 1 range and shouldn't be checked
+        not_quite_probabilities = \
+            set(get_variables_transitively(self._model, (OXMETA, 'not_a_probability_even_though_it_should_be')))
+
+        # remove not_a_probability_even_though_it_should_be annotation to prevent this being used for naming variables
```

Comment:
@mirams On paper I do agree, however that is diffuclt to implemnt as it is because `hodgkin_huxley_gate` is annotated as being allowed multiple times, which would mean it would also remove the names of all the gates from being used, as these all inherit this property.

Is that intentional? Can you have multiple `membrane_L_type_calcium_current_f_gate` e.g. in the same model so we shouldn't use that as a name? 

Or did you mean that we could have different gates, but each gate themselves is unique? In which case we should probably remove the allowed multiple times annotation.

Reply:
I guess there are not many things that are an Annotation - thing you can drag and drop on the web lab metadata tool - themselves that should be used multiple times. Most things that can be used multiple times are properties that annotations themselves share (is a concentration, is a probability, is an ion current etc.).

What you'd want to ban is variables taking the name of Annotations that "can be used multiple times". i.e. any tag that has both:
```
    a :Annotation ;
    a :MultipleUsesAllowed ;
```
which includes HH_gating_variable, Markov_Model_State, and not-a-probability.

Reply:
and these ones which also makes sense:
<img width="267" alt="image" src="https://user-images.githubusercontent.com/858776/215155297-5a0ddca2-4294-44fb-931e-036871330751.png">
because they will appear multiple times on things in the same model